### PR TITLE
Update CLI docs and create reproducible process to update these docs.

### PR DESCRIPTION
**Pull request by geeknoid**
Staging: https://geeknoid.github.io/istio.github.io/docs/reference/commands/

### CODE REVIEW

File: scripts/grab_reference_docs.sh [hunk 1]
```diff
@@ -39,12 +41,37 @@ locate_file() {
...
+# runs it to extract its command-line docs
+#
+# TODO: Even though this CDs into the source tree we've extracted, it's not actually
+# using that as input sources since imports are resolved through $GOPATH and such.
+# I'm not clear what voodoo is needed so I'm leaving this as-is for the time being
+get_command_doc() {
+    COMMAND_PATH=$1
+    COMMAND=$2
+
+    cd $COMMAND_PATH
```
Comment (ZackButcher) [inline scripts/grab_reference_docs.sh]: pushd/popd would be better so we can exit this function back in the directory it was invoked from.

Not blocking.

**Comment by ldemailly** [comment]
istioctl gen command table isn’t rendering on the preview

also is there a way to extract the common flags and list them once rather than for each subcommand ?

there also seems to be leftover flag from glog (-v, -logmodule etc)

**Comment by geeknoid** [comment]
I see istioctl gen-deploy listed, which is what I see when doing "istioctl --help".

Yes, there's a lot of glog crap left behind. I've got a plan to get rid of those (using the istio/glog repo). But for those, those are part of the command-line interfaces of our commands.

As for the layout, we can definitely update the layout. The logic that controls the layout is in istio/pkg/collateral/control.go. We can tackle that separately.

**Comment by ldemailly** [comment]
as mentioned above
https://istio.io/docs/reference/commands/istioctl.html

**Comment by ldemailly** [comment]
gen-deploy section is broken

**Comment by geeknoid** [comment]
I pushed a quick fix for that. The real fix will involve updates to the
markdown production logic in istio/pkg/collateral.

**Comment by ldemailly** [comment]
also all the argument types are missing ( < > needs escaping to &lt; I am guessing)

**Comment by geeknoid** [comment]
Huh, I was expecting this to be taken of by markdown conversion. I guess
I'll need to emit entities...

**Comment by ldemailly** [comment]
maybe there is an option to the table for md to do the escaping for us, not sure

**Comment by ldemailly** [comment]
the upside I guess is you could use <i>italic</i> etc... to make it more lively

**Pull request merged**